pkg/p2p/libp2p/internal/handshake/handshake.go

```diff
@@ -149,6 +150,16 @@ func (s *Service) Handshake(ctx context.Context, stream p2p.Stream, peerMultiadd
 		return nil, ErrInvalidSyn
 	}
 
+	observedUnderlayAddrInfo, err := libp2ppeer.AddrInfoFromP2pAddr(observedUnderlay)
+	if err != nil {
+		return nil, fmt.Errorf("extract addr from P2P: %w", err)
+	}
+
+	if s.libp2pID != observedUnderlayAddrInfo.ID {
+		//NOTE eventually we will return error here, but for now we want to gather some statistics
+		s.logger.Warningf("received peer ID %s does not match ours: %s", observedUnderlayAddrInfo.ID, s.libp2pID)
```

Comment:
@acud normally we should return error here but we thought about maybe leaving it as a warning to gather some statistics about how often it happens.
Do you think this kind of data would be valuable or should we error here right away?

Reply:
sure :+1: